### PR TITLE
Fix CORS on Android for WebDAV

### PR DESCRIPTION
**Pull request by penn5**
See https://github.com/johannesjo/super-productivity-android/pull/17 for the relevant API.

# Description

This is needed to fix CORS in WebDAV sync in the Android app

## Issues Resolved

https://github.com/johannesjo/super-productivity/issues/1847

## Check List

- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable.

I haven't tested this on the stable version of the Android app but in theory it should work.

### CODE REVIEW

**Comment by johannesjo** [comment]
This is great! Thank you very much! This also opens up the door for making Jira work on mobile android.

File: src/app/features/android/android-interface.ts [hunk 1]
```diff
@@ -56,6 +56,28 @@ export interface AndroidInterface {
...
+    rId: string,
+    url: string,
+    method: string,
+    data: string,
+    username: string,
+    password: string,
+    readResponse: boolean,
+  ): void;
+
+  makeHttpRequestCallback(rId: string, result: Object): void;
```
Comment (johannesjo) [inline src/app/features/android/android-interface.ts:79]: The linting doesn't like object. If you're sure that this is an object you can use something like 
```
{[key: string]: string | number}
``` 
Or possibly `unknown` or even `any`

Comment (penn5) [reply src/app/features/android/android-interface.ts:79]: Will `{}` work? I haven't used TS before 😅

Comment (johannesjo) [reply src/app/features/android/android-interface.ts:79]: No unfortunately not :) The quickest compromise might be:
```
{[key: string]:  any }
```

**Comment by penn5** [comment]
Uh, you do know the lint failed right?

**Comment by johannesjo** [comment]
@penn5 yes, but I decided it's quicker, if I just fix it locally :)

**Pull request merged**